File: api-server/models/Track.js
const mongoose = require('mongoose')
const Artist = require('../models/Artist')
const Album = require('../models/Album')

const trackSchema = mongoose.Schema({
    _id: String,
    album : {
        type: String,
        ref: 'Album'
    },
    artists : {
        type: [String],
        ref: Artist
    },
    available_markets : [String],
    disc_number : Number,
    duration_ms : Number,
    explicit : Boolean,
    external_ids: { 
        isrc: String
    },
    external_urls: { 
        spotify: String
    },
    href : String,
    is_local : Boolean,
    name : String,
    popularity : Int,
    preview_url : String,
    track_number : Int,
    type : String,
    uri : String
}, { versionKey: false })

module.exports = mongoose.model('Track', trackSchema)

File: app/src/main/java/com/lucianghimpu/matchmefy/data/dataModels/Track.kt
package com.lucianghimpu.matchmefy.data.dataModels

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Track (
    val album : Album,
    val artists : List<Artist>,
    val external_urls : ExternalUrls,
    val id : String,
    val name : String,
    val preview_url : String,
    val uri : String
) : Parcelable